src/main/java/core/basesyntax/service/impl/FileReaderImpl.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax.service.impl;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FileReaderImpl implements core.basesyntax.service.FileReader {
+
+    @Override
+    public List<String> read(String fileName) {
+        List<String> list = new ArrayList<>();
+
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fileName))) {
+            String line = bufferedReader.readLine();
+            while (line != null) {
+                list.add(line);
+                line = bufferedReader.readLine();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can not read from file " + fileName);
```

Comment:
When rethrowing the exception as a RuntimeException, it's important to include the original exception to preserve the stack trace. Please pass the caught exception 'e' as a second argument to the RuntimeException constructor.